component/mcsd/component.go

```diff
@@ -3,28 +3,47 @@ package mcsd
 import (
 	"context"
 	"encoding/json"
-	"errors"
 	"fmt"
 	"net/http"
 	"net/url"
+	"slices"
 	"strings"
+	"sync"
 
 	fhirclient "github.com/SanteonNL/go-fhir-client"
 	"github.com/nuts-foundation/nuts-knooppunt/component"
+	"github.com/nuts-foundation/nuts-knooppunt/lib/coding"
 	"github.com/rs/zerolog/log"
 	"github.com/zorgbijjou/golang-fhir-models/fhir-models/fhir"
 )
 
 var _ component.Lifecycle = &Component{}
 
+var rootDirectoryResourceTypes = []string{"Organization", "Endpoint"}
+var directoryResourceTypes = []string{"Organization", "Endpoint", "Location", "HealthcareService"}
+
+// Component implements a mCSD Update Client, which synchronizes mCSD FHIR resources from remote mCSD Directories to a local mCSD Directory for querying.
+// It is configured with a root mCSD Directory, which is used to discover organizations and their mCSD Directory endpoints.
+// Organizations refer to Endpoints through Organization.endpoint references.
+// Synchronization is a 2-step process:
+// 1. Query the root mCSD Directory for Organization resources and their associated Endpoint resources of type 'mcsd-directory-endpoint'.
+// 2. For each discovered mCSD Directory Endpoint, query the remote mCSD Directory for its resources and copy them to the local mCSD Query Directory.
+//   - The following resource types are synchronized: Organization, Endpoint, Location, HealthcareService
+//   - An organization's mCSD Directory may only return Organization resources that:
+//   - exist in the root mCSD Directory (link by identifier, name must be the same)
+//   - have the same mcsd-directory-endpoint as the directory being queried
+//   - These are mitigating measures to prevent an attacker to spoof another care organization.
 type Component struct {
 	config       Config
 	fhirClientFn func(baseURL *url.URL) fhirclient.Client
+
+	adminDirectories    []administrationDirectory
```

Comment:
probably a personal pref, but I like `administrationDirectory` over `adminDirectory` as admin is ambiguous